ui/app/components/clients/date-range.ts

```diff
@@ -132,17 +146,17 @@ export default class ClientsDateRangeComponent extends Component<Args> {
       start_time: undefined,
       end_time: undefined,
     };
-    if (this.startDate) {
-      const [year, month] = this.startDate.split('-');
-      if (year && month) {
-        params.start_time = formatDateObject({ monthIdx: parseInt(month) - 1, year: parseInt(year) }, false);
-      }
+    this.startDate = this.modalStart;
+    this.endDate = this.modalEnd;
+
+    let year, month;
+    [year, month] = this.startDate.split('-');
+    if (year && month) {
+      params.start_time = formatDateObject({ monthIdx: parseInt(month) - 1, year: parseInt(year) }, false);
```

Comment:
Ensuring that `parseInt` uses base 10. If you don't supply the second param (the radix) the method infers it from the first value, which can potentially lead to unexpected results ex: if we allowed `0x` as the string this would be considered hex and thus 16 would be the radix deferred value. It's for sure on the cautious side, but easy to account for.
```suggestion
      params.start_time = formatDateObject({ monthIdx: parseInt(month, 10) - 1, year: parseInt(year, 10) }, false);
```

Reply:
Good catch!